lib/install/management/dispatcher.go

```diff
@@ -165,13 +192,19 @@ func (d *Dispatcher) Dispatch(conf *metadata.VirtualContainerHostConfigSpec, set
 	if err = d.removeApplianceIfForced(conf); err != nil {
 		return errors.Errorf("%s", err)
 	}
+
 	if err = d.createAppliance(conf, settings); err != nil {
 		return errors.Errorf("Creating the appliance failed with %s. Exiting...", err)
 	}
+
 	if err = d.uploadImages(settings.ImageFiles); err != nil {
 		return errors.Errorf("Uploading images failed with %s. Exiting...", err)
 	}
 
+	if err = d.registerExtension(conf, settings); err != nil {
```

Comment:
per my last comment to myself, this should be registerExtension(conf, settings.Extension)